sympy2jax/sympy_module.py

```diff
@@ -187,6 +187,28 @@ def sympy(self, memodict: dict, func_lookup: dict) -> sympy.Expr:
         # memodict not needed as sympy deduplicates internally
         return sympy.Integer(self._numerator) / sympy.Integer(self._denominator)
 
+class _Constant(_AbstractNode):
+    _value: jnp.ndarray
+    _expr: sympy.Expr
+
+    _constant_lookup = {
+        sympy.E: jnp.e,
+        sympy.pi: jnp.pi,
+        sympy.EulerGamma: jnp.euler_gamma,
+        sympy.I: 1j,
+    }
+    def __init__(self, expr: sympy.Expr, make_array: bool):
+        assert expr in (sympy.E, sympy.pi, sympy.EulerGamma, sympy.I)
```

Comment:
Nit: these four values are duplicated several times. I think it would be better to move `_constant_lookup` to the global scope, and then do e.g. `assert expr in _constant_lookup.keys()`. This avoids the possibility of each case going out of sync with another.

Reply:
Agreed, that's better

Reply:
I think you could even skip the `.keys()` part if you wanted, just `assert expr in _constant_lookup` should be fine